components/brave_wallet_ui/components/desktop/views/explore_web3/explore_web3.tsx

```diff
@@ -171,12 +166,9 @@ export const ExploreWeb3View = () => {
     selectedCategory === null
 
   // methods
-  const onDappClick = React.useCallback(
-    (dappId: number) => {
-      history.push(makeDappDetailsRoute(dappId.toString()))
-    },
-    [history]
-  )
+  const onDappClick = React.useCallback((dapp: BraveWallet.Dapp) => {
+    setSelectedDapp(dapp)
+  }, [])
```

Comment:
This callback method is not needed since it only calls `setSelectedDapp` with unmodified args. Please use `setSelectedDapp` directly